actions/rules/action_state_to_metastate.go

```diff
@@ -0,0 +1,472 @@
+package rules
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/fabric8-services/fabric8-wit/actions/change"
+
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+// ActionStateToMetaState implements the bidirectional mapping between state and column.
+type ActionStateToMetaState struct {
+	Db     application.DB
+	Ctx    context.Context
+	UserID *uuid.UUID
+}
+
+// make sure the rule is implementing the interface.
+var _ Action = ActionStateToMetaState{}
+
+func (act ActionStateToMetaState) containsUUID(s []uuid.UUID, e uuid.UUID) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (act ActionStateToMetaState) contains(s []interface{}, e interface{}) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (act ActionStateToMetaState) removeElement(s []interface{}, e interface{}) []interface{} {
```

Comment:
I cannot find a test for this function. As simple as it looks this is really dangerous. First, it can be wrong. Second, I think you're leaking memory here. See this for an explanation: 

> NOTE If the type of the element is a pointer or a struct with pointer fields, which need to be garbage collected, the above implementations of Cut and Delete have a potential memory leak problem: some elements with values are still referenced by slice a and thus can not be collected. 

(source https://github.com/golang/go/wiki/SliceTricks#delete-without-preserving-order)

Reply:
Fixed. 35e603400b4769be888722b5a75f18c0c9aad457

Reply:
I've commented on the revision 35e6034. The issue is different now. The functions `removeElement` and `contains` should not be member functions.